dashboard/modules/dashboard_sdk.py

```diff
@@ -135,8 +135,19 @@ def parse_cluster_info(
 ) -> ClusterInfo:
     module_string, inner_address = _split_address(address)
 
-    # If user passes http(s):// or ray://, go through normal parsing.
-    if module_string in {"http", "https", "ray"}:
+    # If user passes in ray://, raise error. Cluster-level submission should
+    # not use a Ray client address.
+    if module_string == "ray":
+        raise ValueError(
+            f'Got "{address}", which is a Ray client address. '
+            "Cluster-level submission should not use a Ray "
+            "client address. If you are not sure where this address "
+            'comes from, please check the "RAY_ADDRESS" '
+            "environment variable to see if it is set."
+        )
```

Comment:
I think we can improve this error message to make it more clear / actionable. How about something like this:

`Got a Ray client address {address}, but you're trying to connect to the dashboard SDK. This requires the address of the HTTP address of the Ray dashboard server. If you didn't pass this address explicitly, it may be set in the RAY_ADDRESS environment variable.`

Reply:
Most users who encounter this probably absentmindedly put `ray://` instead of `http://`, so it might be worth specifically including `ray://` somewhere in the error message (it won't be obvious to the user that "ray client address" means "ray:// address".)  Also I've seen this linked around for error message guidelines, it's generally useful: https://spark.apache.org/error-message-guidelines.html

Reply:
Thanks for the link and the feedback @architkulkarni @edoakes. I've updated the error message.